npm_and_yarn/lib/dependabot/npm_and_yarn/update_checker.rb

```diff
@@ -189,7 +193,8 @@ def top_level_dependency_lookup
 
       def build_updated_dependency(update_details)
         original_dep = update_details.fetch(:dependency)
-        version = update_details.fetch(:version).to_s
+        removed = update_details.fetch(:removed, false)
+        version = update_details.fetch(:version).to_s unless removed
```

Comment:
🔎 When I read this, I read that `version` is `nil` when `dependency.removed?` is `true`. Is that always the case? Could the absence of a version suggest removal?

Reply:
At this stage of the code, yes. However, it is valid for a `Dependency` to have just a requirement range (^6.0.0) and no specific version when we are parsing the dependency files and in those cases I didn't want them to be flagged as removed.